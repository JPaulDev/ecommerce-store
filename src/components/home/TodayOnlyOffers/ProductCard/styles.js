import styled, { css } from 'styled-components';

const flexStyles = css`
  align-items: center;
  display: flex;
  justify-content: center;
`;

const textStyles = css`
  -webkit-box-orient: vertical;
  display: -webkit-box;
  overflow: hidden;
  text-align: center;
`;

export const ListItem = styled.li`
  background-color: white;
  border: 1px solid var(--color-grey-9);
  border-radius: 10px;
  flex-direction: column;
  height: 100%;
  padding: 20px;
  ${flexStyles}

  img {
    height: auto;
    width: 150px;
  }

  @media screen and (min-width: 570px) {
    min-height: 510px;
  }

  @media screen and (max-width: 1109.98px) {
    :nth-child(4n) {
      display: none;
    }
  }
`;

export const ProductName = styled.p`
  -webkit-line-clamp: 2;
  margin: 5px 0 10px;
  ${textStyles}

  @media screen and (min-width: 570px) {
    margin: 5px 0 auto;
  }
`;

export const ProductDescription = styled.p`
  -webkit-line-clamp: 3;
  color: var(--color-grey-8);
  font-size: 0.88rem;
  margin-top: 10px;
  ${textStyles}
`;

export const Wrapper = styled.div`
  border-bottom: 2px solid #d2d6d6;
  margin: 5px 0;
  padding-bottom: 10px;
  width: 100%;
  ${flexStyles}
`;
