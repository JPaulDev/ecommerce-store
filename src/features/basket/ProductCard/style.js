import styled, { css } from 'styled-components';

const overflowEllipsisMixin = css`
  -webkit-box-orient: vertical;
  display: -webkit-box;
  overflow: hidden;
`;

export const buttonMixin = css`
  background-image: linear-gradient(
    to bottom,
    ${({ theme }) => theme.colors.green[500]},
    ${({ theme }) => theme.colors.green[700]}
  );
  border-radius: ${({ theme }) => theme.borderRadius.lg};
  box-shadow: ${({ theme }) => theme.boxShadows.md};
  color: white;
  cursor: pointer;
  font-weight: 700;
  margin-top: 10px;
  padding: 13px;
  font-size: ${({ theme }) => theme.fontSizes[14]};
  text-transform: uppercase;

  &:disabled {
    background: ${({ theme }) => theme.colors.neutral[300]};
    cursor: not-allowed;
    box-shadow: none;
  }

  &:not(:disabled):hover {
    box-shadow: inset 0 0 0 100px rgb(0 0 0 / 10%),
      ${({ theme }) => theme.boxShadows.hover};
  }

  &:not(:disabled):active {
    transform: scale(0.96);
  }
`;

export const HorizontalCard = styled.li`
  --border-color: ${({ theme }) => theme.colors.stone[400]};

  border: 1px solid var(--border-color);
  display: grid;
  grid-template-columns: auto 1fr;
  padding: 20px 25px;

  > img {
    height: auto;
    width: 120px;
  }

  @media ${({ theme }) => theme.breakpoints.mobileLg} {
    > img {
      width: 150px;
    }
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    grid-template-columns: auto 1fr 280px;
  }

  @media ${({ theme }) => theme.breakpoints.desktopXs} {
    > img {
      width: 180px;
    }
  }
`;

export const LeftContainer = styled.div`
  padding: 10px 0 10px 25px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  row-gap: 10px;

  > img {
    height: auto;
    width: 70px;
  }

  .sale-tag {
    background-color: ${({ theme }) => theme.colors.red[500]};
    border-radius: ${({ theme }) => theme.borderRadius.lg};
    color: white;
    font-size: ${({ theme }) => theme.fontSizes[14]};
    font-weight: 600;
    padding: 3px 10px;
  }

  > p {
    color: ${({ theme }) => theme.colors.stone[600]};
    font-size: ${({ theme }) => theme.fontSizes[15]};
    line-height: 1.5;
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    padding: 10px 25px;
  }

  @media ${({ theme }) => theme.breakpoints.desktopXs} {
    > img {
      width: 90px;
    }
  }
`;

export const RightContainer = styled.div`
  border-top: 2px solid var(--border-color);
  grid-column: 1 / 3;
  padding-top: 10px;
  display: flex;
  justify-content: space-between;
  margin-top: 10px;

  .price-container {
    display: flex;
    flex-direction: column;
    position: relative;
  }

  button {
    ${buttonMixin}
    min-width: 160px;
  }

  .container {
    align-items: flex-end;
    flex-direction: column;
    display: flex;
    justify-content: space-between;
    margin-top: 10px;
    row-gap: 5px;
  }

  @media ${({ theme }) => theme.breakpoints.tabletSm} {
    margin-left: 25px;
    grid-column: 2 / 3;
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    padding-left: 25px;
    padding-top: 0;
    margin: 0;
    border-left: 1px solid var(--border-color);
    flex-direction: column;
    justify-content: space-evenly;
    grid-column: auto;
    border-top: none;

    .quantity-wrapper {
      position: absolute;
      right: 0;
      top: 0;
    }

    button {
      width: 100%;
    }

    .container {
      flex-direction: row;
      align-items: center;
    }
  }
`;

export const VerticalCard = styled.li`
  align-items: center;
  display: flex;
  justify-content: center;
  background-color: white;
  border: 1px solid ${({ theme }) => theme.colors.stone[200]};
  border-radius: ${({ theme }) => theme.borderRadius.xl};
  flex-direction: column;
  padding: 20px;
  text-align: center;

  > img {
    height: auto;
    width: 150px;
  }

  .product-name {
    ${overflowEllipsisMixin}
    -webkit-line-clamp: 2;
  }

  .product-description {
    ${overflowEllipsisMixin}
    color: ${({ theme }) => theme.colors.stone[600]};
    font-size: ${({ theme }) => theme.fontSizes[14]};
    -webkit-line-clamp: 3;
    line-height: 1.5;
    margin: 10px 0;
  }

  .divider {
    height: 2px;
    background-color: ${({ theme }) => theme.colors.stone[300]};
    margin: 10px 0;
    width: 100%;
  }

  > button {
    ${buttonMixin}
    min-width: 180px;
  }

  @media ${({ theme }) => theme.breakpoints.mobileLg} {
    min-height: 520px;
  }
`;
