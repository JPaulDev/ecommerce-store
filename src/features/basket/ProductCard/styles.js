import styled, { css } from 'styled-components';

const flexMixin = css`
  align-items: center;
  display: flex;
  justify-content: center;
`;

const overflowEllipsisMixin = css`
  -webkit-box-orient: vertical;
  display: -webkit-box;
  overflow: hidden;
  text-align: center;
`;

const imageMixin = css`
  height: auto;
  max-width: 100%;
`;

const containerMixin = css`
  display: flex;
  justify-content: space-between;
  margin-top: 10px;
`;

export const ListItem = styled.li`
  ${({ variant }) => {
    if (variant === 'horizontal') {
      return css`
        border: 1px solid var(--color-grey-9);
        display: grid;
        grid-template-columns: auto 1fr;
        margin-bottom: 15px;
        padding: 20px 25px;

        @media screen and (min-width: 800px) {
          grid-template-columns: auto 1fr 280px;
        }
      `;
    }

    return css`
      background-color: white;
      border: 1px solid var(--color-grey-12);
      border-radius: 10px;
      flex-direction: column;
      height: 100%;
      padding: 20px;
      ${flexMixin}

      img {
        height: auto;
        width: 150px;
      }

      :nth-child(4n) {
        display: none;

        @media screen and (min-width: 1110px) {
          display: flex;
        }
      }

      @media screen and (min-width: 570px) {
        min-height: 510px;
      }
    `;
  }}
`;

export const ProductName = styled.p`
  -webkit-line-clamp: 2;
  margin: 5px 0 10px;
  ${overflowEllipsisMixin}

  @media screen and (min-width: 570px) {
    margin: 5px 0 auto;
  }
`;

export const ProductDescription = styled.p`
  color: var(--color-grey-4);
  font-size: var(--font-size-14);

  ${({ variant }) => {
    if (variant === 'horizontal') {
      return css`
        margin: 5px 0;
      `;
    }

    return css`
      -webkit-line-clamp: 3;
      line-height: 1.5;
      margin-top: 10px;
      ${overflowEllipsisMixin}
    `;
  }}
`;

export const PartNumberWrapper = styled.div`
  border-bottom: 2px solid var(--color-grey-11);
  margin: 5px 0;
  padding-bottom: 10px;
  width: 100%;
  ${flexMixin}
`;

export const ImageWrapper = styled.div`
  height: auto;
  width: 120px;

  img {
    ${imageMixin}
  }

  @media screen and (min-width: 530px) {
    width: 150px;
    height: auto;
  }

  @media screen and (min-width: 930px) {
    width: 180px;
    height: auto;
  }
`;

export const LeftContainer = styled.div`
  padding: 10px 0 10px 25px;

  @media screen and (min-width: 800px) {
    padding: 10px 25px;
  }
`;

export const LogoWrapper = styled.div`
  height: auto;
  width: 75px;

  img {
    ${imageMixin}
  }

  @media screen and (min-width: 930px) {
    width: 90px;
    height: auto;
  }
`;

export const RightContainer = styled.div`
  border-top: 2px solid var(--color-grey-9);
  grid-column: 1/3;
  padding-top: 10px;
  ${containerMixin}

  @media screen and (min-width: 650px) {
    margin-left: 25px;
    grid-column: 2/3;
  }

  @media screen and (min-width: 800px) {
    padding-left: 25px;
    padding-top: 0;
    margin: 0;
    border-left: 1px solid var(--color-grey-9);
    flex-direction: column;
    justify-content: space-evenly;
    grid-column: initial;
    border-top: none;
  }
`;

export const PriceContainer = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
`;

export const QuantityWrapper = styled.div`
  @media screen and (min-width: 800px) {
    position: absolute;
    right: 0;
    top: 0;
  }
`;

export const ButtonContainer = styled.div`
  width: 160px;

  @media screen and (min-width: 650px) {
    width: 200px;
  }

  @media screen and (min-width: 800px) {
    width: 100%;
  }
`;

export const InnerContainer = styled.div`
  align-items: flex-end;
  flex-direction: column;
  ${containerMixin}

  @media screen and (min-width: 800px) {
    flex-direction: row;
    align-items: center;
  }
`;

export const SaleTag = styled.div`
  background-color: var(--color-red-3);
  border-radius: 5px;
  color: white;
  display: inline-block;
  font-size: var(--font-size-14);
  font-weight: var(--font-weight-semibold);
  margin-top: 5px;
  padding: 2px 10px;
  text-transform: uppercase;
`;
