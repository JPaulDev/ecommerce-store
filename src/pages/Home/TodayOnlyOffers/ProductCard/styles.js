import styled from 'styled-components';

export const Wrapper = styled.div`
  border: 1px solid var(--color-grey-9);
  overflow: hidden;
`;

export const Container = styled.div`
  align-items: center;
  background-color: white;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  min-height: 510px;
  min-width: 240px;
  padding: 20px;
`;

export const ProductImage = styled.img`
  height: 150px;
  width: auto;
`;

export const ProductName = styled.p`
  font-size: 0.94rem;
  font-weight: 600;
  margin: 5px 0 10px;
  text-align: center;
`;

export const ProductDescription = styled.p`
  color: var(--color-grey-8);
  font-size: 0.88rem;
  text-align: center;
`;

export const PartNumberWrapper = styled.div`
  border-bottom: 2px solid #d2d6d6;
  display: flex;
  justify-content: center;
  margin: 5px 0;
  padding-bottom: 13px;
  width: 100%;
`;
