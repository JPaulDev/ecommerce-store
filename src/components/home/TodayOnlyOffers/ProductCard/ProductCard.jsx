import Image from 'next/future/image';
import { useState, forwardRef } from 'react';
import { useDispatch } from 'react-redux';
import { addProduct } from '../../../../features/basket/basketSlice';
import usePrice from '../../../../hooks/usePrice';
import {
  Price,
  PreviousPrice,
  PartNumber,
  QuantityDropdown,
  StockIndicator,
  AddToBasket,
} from '../../../common';
import * as Styled from './styles';

function ProductCard({ product = {} }, ref) {
  const [quantity, setQuantity] = useState(1);
  const { price, salePrice } = usePrice(product);
  const dispatch = useDispatch();
  const isOutOfStock = product.stock <= 0;

  const handleChangeQuantity = (e) => setQuantity(parseInt(e.target.value, 10));
  const handleAddProduct = () => {
    dispatch(addProduct(product, quantity));
    setQuantity(1);
  };

  return (
    <Styled.Wrapper ref={ref} data-testid="product">
      <Styled.Container>
        <Styled.ImageWrapper>
          <Image src={product.image} alt="" quality={85} />
        </Styled.ImageWrapper>
        <Styled.ProductName>{product.name}</Styled.ProductName>
        <Styled.ProductDescription>
          {product.description?.join(', ')}
        </Styled.ProductDescription>
        <Styled.PartNumberWrapper>
          <PartNumber fontSize="0.75rem" sku={product.sku} />
        </Styled.PartNumberWrapper>
        <Price price={salePrice || price} fontSize="1.9rem" />
        {salePrice && <PreviousPrice price={price} fontSize="0.93rem" />}
        <QuantityDropdown
          quantity={quantity}
          isDisabled={isOutOfStock}
          onChangeQuantity={handleChangeQuantity}
        />
        <StockIndicator stock={product.stock} marginTop="8px" />
        <AddToBasket
          isDisabled={isOutOfStock}
          width="160px"
          fontSize="0.75rem"
          onAddProduct={handleAddProduct}
        />
      </Styled.Container>
    </Styled.Wrapper>
  );
}

export default forwardRef(ProductCard);
