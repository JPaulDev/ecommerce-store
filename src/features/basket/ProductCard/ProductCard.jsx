import dynamic from 'next/dynamic';
import Image from 'next/future/image';
import { useState, forwardRef } from 'react';
import { useDispatch } from 'react-redux';
import { addProduct } from '../basketSlice';
import { usePrice } from '../../../hooks';
import {
  Price,
  PreviousPrice,
  PartNumber,
  Quantity,
  StockDisplay,
  AddToBasket,
} from '../../../components/product';
import * as Styled from './styles';

const DeliveryDate = dynamic(
  () => import('../../../components/product/DeliveryDate'),
  {
    ssr: false,
  }
);

function ProductCard({ product, variant = 'horizontal' }, ref) {
  const [quantity, setQuantity] = useState(1);
  const dispatch = useDispatch();

  const { price, previousPrice } = usePrice({
    isOnSale: product.isOnSale,
    discount: product.discount,
    price: product.price,
    currencyCode: product.currencyCode,
  });

  const handleChangeQuantity = (e) => setQuantity(Number(e.target.value));

  const handleAddProduct = () => {
    dispatch(addProduct(product, quantity));
    setQuantity(1);
  };

  return (
    <>
      {variant === 'horizontal' && (
        <Styled.ListItem ref={ref} variant={variant}>
          <Styled.ImageWrapper>
            <Image
              src={product.imageUrl}
              alt=""
              width={200}
              height={200}
              quality={90}
              priority="true"
            />
          </Styled.ImageWrapper>
          <Styled.LeftContainer>
            <Styled.LogoWrapper>
              <Image
                src={product.brand}
                alt=""
                width={110}
                height={24}
                quality={90}
              />
            </Styled.LogoWrapper>
            {product.isOnSale && <Styled.SaleTag>SALE</Styled.SaleTag>}
            <Styled.ProductDescription variant={variant}>
              {product.name}, {product.description}
            </Styled.ProductDescription>
            <PartNumber fontSize={13}>{product.sku}</PartNumber>
          </Styled.LeftContainer>
          <Styled.RightContainer>
            <Styled.PriceContainer>
              <Price fontSize={32}>{price}</Price>
              {product.isOnSale && (
                <PreviousPrice>{previousPrice}</PreviousPrice>
              )}
              <Styled.QuantityWrapper>
                <Quantity
                  quantity={quantity}
                  onChangeQuantity={handleChangeQuantity}
                />
              </Styled.QuantityWrapper>
            </Styled.PriceContainer>
            <Styled.ButtonContainer>
              <AddToBasket
                type="button"
                width="100%"
                fontSize={14}
                onAddProduct={handleAddProduct}
              />
              <Styled.InnerContainer>
                <StockDisplay stockStatus={product.stockStatus} />
                <DeliveryDate stockStatus={product.stockStatus} />
              </Styled.InnerContainer>
            </Styled.ButtonContainer>
          </Styled.RightContainer>
        </Styled.ListItem>
      )}

      {variant === 'vertical' && (
        <Styled.ListItem ref={ref} variant={variant}>
          <Image
            src={product.imageUrl}
            alt=""
            width={200}
            height={200}
            quality={90}
          />
          <Styled.ProductName>{product.name}</Styled.ProductName>
          <Styled.ProductDescription variant={variant}>
            {product.description}
          </Styled.ProductDescription>
          <Styled.PartNumberWrapper>
            <PartNumber fontSize={12}>{product.sku}</PartNumber>
          </Styled.PartNumberWrapper>
          <Price fontSize={30}>{price}</Price>
          {product.isOnSale && <PreviousPrice>{previousPrice}</PreviousPrice>}
          <Quantity
            quantity={quantity}
            isDisabled={product.stockStatus === 0}
            onChangeQuantity={handleChangeQuantity}
          />
          <StockDisplay stockStatus={product.stockStatus} marginTop="8px" />
          <AddToBasket
            isDisabled={product.stockStatus === 0}
            width="180px"
            fontSize={12}
            onAddProduct={handleAddProduct}
          />
        </Styled.ListItem>
      )}
    </>
  );
}

export default forwardRef(ProductCard);
