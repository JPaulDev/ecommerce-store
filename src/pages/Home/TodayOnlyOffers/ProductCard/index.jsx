import { useState, forwardRef } from 'react';
import usePrice from '../../../../hooks/usePrice';
import Price from '../../../../components/Price';
import PreviousPrice from '../../../../components/PreviousPrice';
import PartNumber from '../../../../components/PartNumber';
import QuantityDropdown from '../../../../components/QuantityDropdown';
import StockIndicator from '../../../../components/StockIndicator';
import AddToBasketButton from '../../../../components/AddToBasketButton';
import * as Styled from './styles';

const ProductCard = forwardRef(({ product }, ref) => {
  const [quantity, setQuantity] = useState(1);
  const { price, salePrice } = usePrice(product);

  const handleChangeQuantity = (e) => setQuantity(parseInt(e.target.value, 10));

  return (
    <Styled.Wrapper ref={ref}>
      <Styled.Container>
        <Styled.ProductImage
          src={product.image}
          alt=""
          width={200}
          height={200}
        />
        <Styled.ProductName>{product.name}</Styled.ProductName>
        <Styled.ProductDescription>
          {product.description.join(', ')}
        </Styled.ProductDescription>
        <Styled.PartNumberWrapper>
          <PartNumber $fontSize="0.75rem">{product.sku}</PartNumber>
        </Styled.PartNumberWrapper>
        <Price price={salePrice || price} fontSize="1.9rem" />
        {salePrice && <PreviousPrice price={price} fontSize="0.93rem" />}
        <QuantityDropdown
          quantity={quantity}
          isDisabled={!product.stock}
          onChangeQuantity={handleChangeQuantity}
        />
        <StockIndicator stock={product.stock} marginTop="8px" />
        <AddToBasketButton
          type="button"
          disabled={!product.stock}
          $width="160px"
          $fontSize="0.75rem"
        >
          Add To Basket
        </AddToBasketButton>
      </Styled.Container>
    </Styled.Wrapper>
  );
});

export default ProductCard;
