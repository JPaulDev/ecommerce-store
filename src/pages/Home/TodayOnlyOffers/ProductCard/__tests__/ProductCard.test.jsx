import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import ProductCard from '../index';

const product = {
  sku: 'CPU-INT-10400F',
  stock: 50,
  name: 'Intel Core i5 10400F',
  image: '',
  price: 129.99,
  salePrice: 114.99,
  description: [
    '6 Cores, 12 Threads',
    'S1200 Comet Lake',
    '2.9GHz, 4.3GHz Turbo',
    '12MB Cache',
  ],
};

describe('ProductCard component', () => {
  it('renders a product card with an image and all product information', () => {
    const tree = renderer.create(<ProductCard product={product} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ProductCard component quantity input', () => {
  it('records user input changes quantity', () => {
    render(<ProductCard product={product} />);
    const select = screen.getByLabelText(/quantity/i);

    userEvent.selectOptions(select, '3');

    expect(select).not.toHaveValue('1');
    expect(select).toHaveValue('3');
  });

  it('is enabled when the product is in stock', () => {
    render(<ProductCard product={{ ...product, stock: 1 }} />);

    expect(screen.getByLabelText(/quantity/i)).toBeEnabled();
  });

  it('is disabled when the product is out of stock', () => {
    render(<ProductCard product={{ ...product, stock: 0 }} />);

    expect(screen.getByLabelText(/quantity/i)).toBeDisabled();
  });
});

describe('ProductCard component add to basket button', () => {
  it('is enabled when the product is in stock', () => {
    render(<ProductCard product={{ ...product, stock: 1 }} />);
    const button = screen.getByRole('button', {
      name: /add to basket/i,
    });

    expect(button).toBeEnabled();
  });

  it('is disabled when the product is out of stock', () => {
    render(<ProductCard product={{ ...product, stock: 0 }} />);
    const button = screen.getByRole('button', {
      name: /add to basket/i,
    });

    expect(button).toBeDisabled();
  });
});

describe('ProductCard component stock indicator', () => {
  it('product is displayed as in stock when 31 or more', () => {
    render(<ProductCard product={{ ...product, stock: 31 }} />);

    expect(screen.getByText(/in stock/i)).toBeInTheDocument();
  });

  it('product is displayed as low stock at 30 or fewer', () => {
    render(<ProductCard product={{ ...product, stock: 30 }} />);

    expect(screen.getByText(/low stock/i)).toBeInTheDocument();
  });

  it('product is displayed as out of stock at 0', () => {
    render(<ProductCard product={{ ...product, stock: 0 }} />);

    expect(screen.getByText(/out of stock/i)).toBeInTheDocument();
  });
});

describe('ProductCard component price display', () => {
  it('displays the products base price if no sale price is passed in', () => {
    render(<ProductCard product={{ ...product, price: 100, salePrice: '' }} />);

    expect(screen.getByText('100')).toBeInTheDocument();
    expect(screen.queryByText(/was/i)).not.toBeInTheDocument();
  });
});
