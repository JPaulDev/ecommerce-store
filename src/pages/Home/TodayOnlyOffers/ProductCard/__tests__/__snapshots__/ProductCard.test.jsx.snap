// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component renders a product card with an image and all product information 1`] = `
<div
  className="sc-kDDrLX leTUAJ"
  data-testid="product"
>
  <div
    className="sc-iqcoie kriKnu"
  >
    <img
      alt=""
      className="sc-crXcEl brVwQW"
      height={200}
      src=""
      width={200}
    />
    <p
      className="sc-evZas cDFuoe"
    >
      Intel Core i5 10400F
    </p>
    <p
      className="sc-breuTD jmuFiM"
    >
      6 Cores, 12 Threads, S1200 Comet Lake, 2.9GHz, 4.3GHz Turbo, 12MB Cache
    </p>
    <div
      className="sc-ksZaOG kJnygA"
    >
      <span
        className="sc-eCYdqJ ejHRxl"
      >
        CPU-INT-10400F
      </span>
    </div>
    <span
      className="sc-bczRLJ eYpvHB"
    >
      <span
        className="sc-gsnTZi iKSuxX"
      >
        £
      </span>
      114
      <span
        className="sc-gsnTZi iKSuxX"
      >
        .99
      </span>
    </span>
    <span
      className="sc-dkzDqf dAHzws"
    >
      <span
        className="sc-hKMtZM ffHsYy"
      >
        Was: 
      </span>
      £129.99
    </span>
    <div
      className="sc-jSMfEi eIqODg"
    >
      <label
        hidden={true}
        htmlFor=":r0:"
      >
        Quantity
      </label>
      <select
        className="sc-gKXOVf hegjUa"
        disabled={false}
        id=":r0:"
        name="quantity"
        onChange={[Function]}
        value={1}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
        <option
          value={5}
        >
          5
        </option>
      </select>
      <span
        className="sc-iBkjds hndTfr"
      />
    </div>
    <div
      className="sc-papXJ eHmRKN"
      title="Item is in stock"
    >
      <i
        className="sc-ftvSup cWqIfK material-symbols-outlined"
      >
        inventory
      </i>
      In Stock
    </div>
    <button
      className="sc-jqUVSM mvzBv"
      disabled={false}
      type="button"
    >
      Add To Basket
    </button>
  </div>
</div>
`;
